pkg/kclient/deployments.go

```diff
@@ -18,6 +22,76 @@ func (c *Client) GetDeploymentByName(name string) (*appsv1.Deployment, error) {
 	return deployment, err
 }
 
+// getDeploymentCondition returns the condition with the provided type
+// from https://github.com/kubernetes/kubectl/blob/8bc20f428d7d5aed031de5fa160081de7b5af2b0/pkg/util/deployment/deployment.go#L58
+func getDeploymentCondition(status appsv1.DeploymentStatus, condType appsv1.DeploymentConditionType) *appsv1.DeploymentCondition {
+	for i := range status.Conditions {
+		c := status.Conditions[i]
+		if c.Type == condType {
+			return &c
+		}
+	}
+	return nil
+}
+
+// WaitForDeploymentRollout waits for deployment to finish rollout. Returns the state of the deployment after rollout.
+func (c *Client) WaitForDeploymentRollout(deploymentName string) (*appsv1.Deployment, error) {
+	glog.V(4).Infof("Waiting for %s deployment roll out", deploymentName)
+
+	w, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Watch(metav1.ListOptions{FieldSelector: "metadata.name=" + deploymentName})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch deployment")
+	}
+	defer w.Stop()
+
+	success := make(chan *appsv1.Deployment)
+	failure := make(chan error)
+
+	go func() {
+		defer close(success)
+		defer close(failure)
+
+		for {
+			val, ok := <-w.ResultChan()
+			if !ok {
+				failure <- errors.New("watch channel was closed")
+				return
+			}
+			//based on https://github.com/kubernetes/kubectl/blob/9a3954bf653c874c8af6f855f2c754a8e1a44b9e/pkg/polymorphichelpers/rollout_status.go#L66-L91
+			if deployment, ok := val.Object.(*appsv1.Deployment); ok {
+				if deployment.Generation <= deployment.Status.ObservedGeneration {
+					cond := getDeploymentCondition(deployment.Status, appsv1.DeploymentProgressing)
+					if cond != nil && cond.Reason == "ProgressDeadlineExceeded" {
```

Comment:
`ProgressDeadlineExceeded` make this a constant maybe?

Reply:
done